ebuild-maintenance/git/text.xml

```diff
@@ -0,0 +1,322 @@
+the historical commits after the initial git commit.
+</p>
+
+</body>
+</subsection>
+
+</body>
+</section>
+
+<section>
+<title>Committing to gentoo.git</title>
+<body>
+
+<subsection>
+<title>Committing and verifying commits</title>
+<body>
+
+<p>
+The recommended way of committing to the Gentoo repository is to use <c>repoman
+commit</c>. It automatically performs the necessary QA checks on the package
+being committed and has other quirks helping with the Gentoo workflow. However,
+it is currently limited to creating a single commit to a single package.
+</p>
+
+<p>
+For any other use case, <c>git commit</c> and other git commands need to be
+used. The valid uses of git include:
+</p>
+
+<ul>
+<li>creating commits spanning across multiple packages and/or multiple areas
+of Gentoo repository (eclasses, licenses, profiles…),</li>
+<li>amending a commit created via <c>repoman commit</c> with additional files
+or fixups,</li>
+<li>combining multiple commits created via <c>repoman commit</c> via <c>git
+rebase</c>.</li>
+</ul>
+
+<p>
+Whenever repoman is not used to commit, you need to manually verify that all
+packages affected by the commit using <c>repoman full</c>. Since repoman is not
+aware of staged changes, please make sure that all files are included
+in the commit.
+</p>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Atomic commits</title>
+<body>
+
+<p>
+Whenever possible, use atomic commits. Try to split your changes into logical
+commits, abiding the following three rules as much as possible:
+</p>
+
+<ol>
+<li>
+Do not include multiple irrelevant changes in a single commit. However, make
+sure not to split relevant changes unnecessarily. For example, if a version bump
+requires changes in the ebuild, it is correct to perform them in a single
+commit. However, if you are fixing an additional bug that has been present
+in the previous version, the fix belongs in a separate commit.
+</li>
+
+<li>
+Split commits at logical unit boundaries. When updating multiple packages,
+preferably use a single commit for each package. Avoid combining changes
+to ebuilds, eclasses, licenses, profiles etc. in a single commit. However,
+do not split relevant or interdependent changes within a single package.
+</li>
+
+<li>
+Avoid creating commits introducing a temporary breakage. Unless impossible,
+add packages in dependency install order. Add licenses before the packages
+needing them. Commit <path>package.mask</path> and other profile changes before
+ebuilds relying on them. Usually it is also acceptable to include those changes
+along with the commit adding the package.
+</li>
+</ol>
+
+<p>
+Please note that revision bumps count as side effects of the changes requiring
+them and do not belong in separate commits. When doing multiple irrelevant
+changes that require a revision bump, it is only necessary to bump the revision
+in the first commit in the series introduced by a single push.
```

Comment:
I am personally against this. I think each commit should move the tree from one stable state to another and it has been this way so far. A user should be able to pick any commit without requiring the knowledge of the push boundaries and expect a "stable" tree state. Has this been discussed on gentoo-dev or is there a QA team statement?

Reply:
Yes, this has been debated to death (around one of the GLEP reviews), with the result of *1:n* where *n* was a fair number of people who agreed with this specific policy, and 1 was against it. So that makes 2. And yes, it's part of GLEP 66.

Long story short, revbumps are only a matter of enforced upgrade on `emerge -... @world`. If the user is doing the handiwork of picking specific commits, it is extremely unlikely he will be doing repetitive `@world` upgrades on successive commits — he is usually just testing stuff and reinstalling a specific package.

Reply:
I retract my opposition since GLEP 66 has been officially approved.